src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -1,66 +1,171 @@
+                return global::WinRT.ExceptionHelpers.GetHRForException(__exception__);
+            }
+            return 0;
+        }
+
+        internal static unsafe Delegate GetValueDelegateForAbi()
+        {
+            if (typeof(T) == typeof(int) ||
+                typeof(T) == typeof(byte) ||
+                typeof(T) == typeof(bool) ||
+                typeof(T) == typeof(sbyte) ||
+                typeof(T) == typeof(short) ||
+                typeof(T) == typeof(ushort) ||
+                typeof(T) == typeof(char) ||
+                typeof(T) == typeof(uint) ||
+                typeof(T) == typeof(long) ||
+                typeof(T) == typeof(ulong) ||
+                typeof(T) == typeof(float) ||
+                typeof(T) == typeof(double) ||
+                typeof(T) == typeof(Guid) ||
+                typeof(T) == typeof(global::System.TimeSpan) ||
+                typeof(T) == typeof(global::Windows.Foundation.Point) ||
+                typeof(T) == typeof(global::Windows.Foundation.Rect) ||
+                typeof(T) == typeof(global::Windows.Foundation.Size) ||
+                typeof(T) == typeof(global::System.Numerics.Matrix3x2) ||
+                typeof(T) == typeof(global::System.Numerics.Matrix4x4) ||
+                typeof(T) == typeof(global::System.Numerics.Plane) ||
+                typeof(T) == typeof(global::System.Numerics.Quaternion) ||
+                typeof(T) == typeof(global::System.Numerics.Vector2) ||
+                typeof(T) == typeof(global::System.Numerics.Vector3) ||
+                typeof(T) == typeof(global::System.Numerics.Vector4) ||
+                (typeof(T).IsEnum && Enum.GetUnderlyingType(typeof(T)) == typeof(int)) ||
+                (typeof(T).IsEnum && Enum.GetUnderlyingType(typeof(T)) == typeof(uint)))
+            {
+                return (Nullable_Delegates.GetValueDelegateAbi)Do_Abi_get_Value_0_Blittable;
+            }
+
+            if (typeof(T) == typeof(global::System.DateTimeOffset))
+            {
+                return (Nullable_Delegates.GetValueDelegateAbiDateTimeOffset)Do_Abi_get_Value_0_DateTimeOffset;
+            }
+
+#if NET
+            // Only when not on NAOT, use LINQ expressions to get the delegate type.
+            // This is not safe on AOT, so in that case we just can't get a delegate.
+            if (RuntimeFeature.IsDynamicCodeCompiled)
+#endif
+            {
+                Nullable<T>.Vftbl.get_Value_0_Type ??= Projections.GetAbiDelegateType(typeof(void*), Marshaler<T>.AbiType.MakeByRefType(), typeof(int));
+
+                return Delegate.CreateDelegate(
+                    Nullable<T>.Vftbl.get_Value_0_Type,
+                    typeof(BoxedValueIReferenceImpl<T>).GetMethod(nameof(Do_Abi_get_Value_0), BindingFlags.NonPublic | BindingFlags.Static)!.MakeGenericMethod(Marshaler<T>.AbiType));
+            }
+
+            throw new NotSupportedException("Failed to get the marshalling delegate for BoxedValueIReferenceImpl`1.");

```

Comment:
Yea this should be fine here.  For all the other types, we should have handled them [here](https://github.com/microsoft/CsWinRT/blob/ae695510af515a373bc010cf46aa188736b2b8de/src/WinRT.Runtime/ComWrappersSupport.cs#L204C46-L212) using `StructTypeDetails` assuming you are on new CsWinRT and its source generator.  Maybe add to the error message the type name for which we failed.

Reply:
Done in https://github.com/microsoft/CsWinRT/pull/1429/commits/71b246a3a7ce36e37abab4e841876c5e68826df1 🙂